### DISCUSSION TITLE
The Synology deploy-hook does not take the name of the certificat anymore

### DISCUSSION THREAD

**Discussion by Mic13710** [acmesh-official/acme.sh#4504] [Q&A] [answered]
Hello,
Since long, I successfully renew my certificat on a docker session installed on my Synology NAS.

On February 2, my LE certificate was successfully renewed, but was not deployed. acme.sh version: 3.0.6 (first edition).
By consulting the log, I realized that the name of my certificate was not taken into account for this deployment, which prevented the deployment from taking place.

`[Thu Feb  2 00:04:42 UTC 2023] _deployApi='/root/.acme.sh/deploy/synology_dsm.sh'

[Thu Feb  2 00:04:42 UTC 2023] _cdomain='ndd.net'

[Thu Feb  2 00:04:42 UTC 2023] SYNO_Certificate   **'THE NAME IS MISSING HERE'**

[Thu Feb  2 00:04:42 UTC 2023] _base_url='http://172.17.0.1:5000'

[Thu Feb  2 00:04:42 UTC 2023] Getting API version

[Thu Feb  2 00:04:42 UTC 2023] GET

[Thu Feb  2 00:04:42 UTC 2023] url='http://172.17.0.1:5000/webapi/query.cgi?api=SYNO.API.Info&version=1&method=query&query=SYNO.API.Auth'

[Thu Feb  2 00:04:42 UTC 2023] timeout=

[Thu Feb  2 00:04:42 UTC 2023] _CURL='curl --silent --dump-header /acme.sh/http.header  -L  -g '

[Thu Feb  2 00:04:42 UTC 2023] ret='0'

[Thu Feb  2 00:04:42 UTC 2023] Logging into 172.17.0.1:5000

[Thu Feb  2 00:04:42 UTC 2023] POST

[Thu Feb  2 00:04:42 UTC 2023] _post_url='http://172.17.0.1:5000/webapi/auth.cgi?enable_syno_token=yes'

[Thu Feb  2 00:04:42 UTC 2023] _CURL='curl --silent --dump-header /acme.sh/http.header  -L  -g '

[Thu Feb  2 00:04:44 UTC 2023] _ret='0'

[Thu Feb  2 00:04:44 UTC 2023] token='RabpfZzxqvW0Q'

[Thu Feb  2 00:04:44 UTC 2023] Getting certificates in Synology DSM

[Thu Feb  2 00:04:44 UTC 2023] POST

[Thu Feb  2 00:04:44 UTC 2023] _post_url='http://172.17.0.1:5000/webapi/entry.cgi'

[Thu Feb  2 00:04:44 UTC 2023] _CURL='curl --silent --dump-header /acme.sh/http.header  -L  -g '

[Thu Feb  2 00:04:44 UTC 2023] _ret='0'

[Thu Feb  2 00:04:44 UTC 2023] escaped_certificate

[Thu Feb  2 00:04:44 UTC 2023] Unable to find certificate:  and $SYNO_Create is not set

[Thu Feb  2 00:04:44 UTC 2023] Error deploy for domain:ndd.net

[Thu Feb  2 00:04:44 UTC 2023] Deploy error.

[Thu Feb  2 00:04:44 UTC 2023] Return code: 1

[Thu Feb  2 00:04:44 UTC 2023] Error renew ndd.net.`

an update of acme.sh was made on February 5 to version 3.0.6 (second edition).
I tried to launch a manual deployment with a scheduled task with this version but the result is always the same: the name of the certificate is not taken into account, which blocks the deployment.
All account.conf and ndd.conf files are correct. The name of the base64 certificate is present in the ndd.conf.
Is this problem known and how to solve it?

**Comment by Mic13710** [comment]
Hello,
I come back with a temporary solution to the deployment failure with the very last version 3.0.6 of acme.sh

1. Removing the "SAVED_" in front of all the lines in the ndd.conf. For instance, SAVED_SYNO_Scheme='http' become SYNO_Scheme='http'. 
2. Changing the name of the certificate from base 64 to its normal name in DSM.
3. When done, launching the deployment. The certificate should be deployed.

However, with this method, all the previously "SAVED_" lines are removed from the new ndd.conf file. 
By chance, I had kept a copy of each file before their change. I have just restored them to their original location, hoping that the problem shall be solved before the next renewal.

An other method is to delete all the "SAVED_" lines in the ndd.conf and to export the same values before launching the deployment.

Export SYNO_Scheme='http'
Export SYNO_Hostname='IP of your NAS in docker'
Export SYNO_Port='Port'
Export SYNO_Username='user'
Export SYNO_Password='Password'
Export SYNO_DID=''
Export SYNO_Certificate='The name of the certificat in DSM'
Export SYNO_TOTP_SECRET='If any'
docker exec Acme sh -c "acme.sh --deploy  -d 'ndd' --deploy-hook synology_dsm"

In that way, the script should rewrite the "SAVED_" lines into the ndd.conf

Hoping it could help others.

**Reply by Mic13710** [reply]
I come back following the launch of the renewal script.
My certificates have been renewed in the last two days. This is OK for the certificates, still not for the deployments because of the base 64 certificate name.
I had to put the names of the certificates in clear in the .conf files and then relaunch the deployments in DSM.
Of course the .conf have been updated with the names again in base 64 by the script. This means that the problem will be the same at the next renewal.
I don't understand why the name of the certificate is saved in base 64 while the identifiers are saved in clear. Wouldn't it be possible to return to the name in clear as it was the case in the first versions?

**Comment by neilpang** [comment, marked as answer]
This was a transition bug in early 3.0.6: the Synology hook renamed `SYNO_Certificate` -> `SYNO_CERTIFICATE` and switched to storing values base64-encoded under `SAVED_` keys, and that first release didn't migrate the old value, so `SYNO_CERTIFICATE` came up empty on renewal -> "Unable to find certificate:".

Current code fixes it -- `deploy/synology_dsm.sh` now runs `_migratedeployconf SYNO_Certificate SYNO_CERTIFICATE "base64"`, so an old config is converted automatically, and it reads/saves `SYNO_CERTIFICATE` (base64) via `SAVED_SYNO_CERTIFICATE`. Upgrade and the first run migrates your conf.

Two notes:
- The env var is now `SYNO_CERTIFICATE` (all caps), not `SYNO_Certificate`.
- Don't hand-edit the `SAVED_*` values -- the cert name is stored base64 on purpose. If your conf is half-edited, just `export SYNO_CERTIFICATE="<name in DSM>"` once and re-run the deploy; it re-saves correctly.
